client/state/site-settings/exporter/selectors.js

```diff
@@ -1,5 +1,127 @@
+/**
+ * Internal dependencies
+ */
+import i18n from 'lib/mixins/i18n';
+
+import moment from 'moment';
+
 import { States } from './constants.js';
 
+function mapOptions( state, section, name, mapFunc ) {
+	const rawData = state.siteSettings.exporter.data.get( 'advancedSettings' );
+	if ( !rawData || rawData.count() === 0 ) return [];
+
+	return rawData.get( section ).get( name ).map( mapFunc );
+}
+
+export function getAuthorOptions( state, section ) {
+	return mapOptions( state, section, 'authors', ( author ) => ( {
+		value: author.get( 'ID' ), label: author.get( 'name' )
+	} ) );
+}
+
+export function getStatusOptions( state, section ) {
+	return mapOptions( state, section, 'statuses', ( status ) => ( {
+		value: status.get( 'name' ), label: status.get( 'label' )
+	} ) );
+}
+
+export function getDateOptions( state, section, endOfMonth ) {
+	return mapOptions( state, section, 'export_date_options', ( date ) => {
+		const year = parseInt( date.get( 'year' ) );
```

Comment:
we should always add the radix to `parseInt` to make it clear what we want with the output

```js
parseInt( date.get( 'year' ), 10 )
```

This is a protection against bugs where the input changes the behavior of `parseInt`